sql/tests: TestMonotonicInserts failed

sql/tests.TestMonotonicInserts [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9988791?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9988791?buildTab=artifacts#/) on release-23.1 @ [b9fe7c37a536ca7138f7bdc1eb16cf1fecda32ed](https://github.com/cockroachdb/cockroach/commits/b9fe7c37a536ca7138f7bdc1eb16cf1fecda32ed):


```
=== RUN   TestMonotonicInserts
--- FAIL: TestMonotonicInserts (21.28s)
=== RUN   TestMonotonicInserts/distsql=on
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/6f2e15d801c52839633054fadf45543b/logTestMonotonicInserts_distsql=on2105527946
    test_log_scope.go:79: use -show-logs to present logs inline
    monotonic_insert_test.go:187: *crdb.MaxRetriesExceededError: retrying txn failed after 50 attempts. original error: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=a0b04325 key=/Table/106/1/863456381903306753/0 pri=0.03818115 epo=0 ts=1683576375.905866123,1 min=1683576375.879605038,0 seq=1} lock=true stat=PENDING rts=1683576375.879605038,0 wto=false gul=1683576376.379605038,0.
    monotonic_insert_test.go:240: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6f2e15d801c52839633054fadf45543b/logTestMonotonicInserts_distsql=on2105527946
    --- FAIL: TestMonotonicInserts/distsql=on (10.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMonotonicInserts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/tests/monotonic_insert_test.go
+++ b/pkg/sql/tests/monotonic_insert_test.go
@@ -30,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
 )
 
 type mtRow struct {
@@ -184,7 +185,12 @@ RETURNING val, sts, node, tb`,
 			l("commit")
 			return nil
 		}); err != nil {
-			t.Errorf("%T: %v", err, err)
+			// Ignore MaxRetriesExceeeded, it can happen under this contended workload.
+			if !errors.HasType(err, (*crdb.MaxRetriesExceededError)(nil)) {
+				t.Errorf("%T: %v", err, err)
+			} else {
+				t.Logf("ignoring %T: %v", err, err)
+			}
 		}
 	}
 

